src/utils.c

```diff
@@ -281,12 +281,42 @@ ts_timestamp_bucket(PG_FUNCTION_ARGS)
 {
 	Interval   *interval = PG_GETARG_INTERVAL_P(0);
 	Timestamp	timestamp = PG_GETARG_TIMESTAMP(1);
+	Timestamp	origin;
```

Comment:
Wondering if this shouldn't be a `Date` so that we only allow shifting by full days?

Reply:
I don't think so. A use case we've encountered is actually using now() as a origin for getting buckets aligned so the last bucket ends in the current time. This makes sure the last bucket is not weirdly half empty.